File: java/0242_ValidAnagram.java
/*
08DEC2025
https://leetcode.com/problems/valid-anagram/
*/

class Solution {
    public boolean isAnagram(String s, String t) {
        if (s.length() != t.length()) {
            return false;
        }

        Map<Character, Integer> charCount = new HashMap<>();
        for (char curr : s.toCharArray()) {
            charCount.put(curr, charCount.getOrDefault(curr, 0) + 1);
        }
        for (char curr : t.toCharArray()) {
            charCount.put(curr, charCount.getOrDefault(curr, 0) - 1);
        }

        for (Integer val : charCount.values()) {
            if (val != 0) {
                return false;
            }
        }

        return true;
        
    }
}

File: csharp/0242_ValidAnagram.cs
/*
20JAN2026
242. Valid Anagram
https://leetcode.com/problems/valid-anagram/

Given two strings s and t, return true if t is an of s, and false otherwise.
*/
public class Solution {
    public bool IsAnagram(string s, string t) {
        if (s.Length != t.Length) {
            return false;
        }

        int[] chars = new int[26];

        foreach (char c in s) {
            chars[c - 'a']++;
        }

        foreach (char c in t) {
            chars[c - 'a']--;
        }

        foreach (int i in chars) {
            if (i != 0) {
                return false;
            }
        }

        return true;
    }
}
